### DISCUSSION TITLE
v0.7.1

### DISCUSSION THREAD

**Discussion by Achronite** [Achronite/mqtt-energenie-ener314rt#84] [Announcements]
# [0.7.1] 2024-02-21

### IMPORTANT: This release requires the following updates that will need to be manually installed first:
* `node.js`: v18.2.0 or greater
* `gpiod` & `libgpiod`: New dependencies that need to be installed  (e.g raspbian: `sudo apt-get gpiod libgpiod`)

### Added

* Support added for MiHome Thermostat (MIHO069), including auto-messaging to obtain telemetry
* Support added for MiHome Click (MIHO089) [#79](https://github.com/Achronite/mqtt-energenie-ener314rt/issues/79)
* MQTT Discovery: Added thermostat, including Climate Control entity
* MQTT Discovery: Added MiHome Click
* The number of retries is now configurable for cached commands (applies to eTRV and thermostat) by setting `cached_retries` in `config,json` (default remains at 10)
* MQTT Discovery: Added 'Battery Timestamp' (mapped to MQTT VOLTAGE_TS), 'Diagnostics Ran' (mapped to DIAGNOSTICS_TS) and 'Valve Exercised (mapped to VALVE_TS) for eTRV
* MQTT Discovery: Added 'Identify' Button for eTRV
* Setting target temperature now caters for 0.5 increments (previously integer)
* A different mechanism of reporting processed commands has been implemented for the thermostat, that relies on the fact that when (and only when) the thermostat processes a command it outputs it's telemetry data.  This mechanism has been used to assume that the command just sent to the device (upon WAKEUP) has been processed successfully; This command has it's retained state set in MQTT [#61](https://github.com/Achronite/mqtt-energenie-ener314rt/issues/61
* MQTT Discovery: A new device `mqtt-energenie-ener314rt` has been added to represent the software and the ENER314-RT board, which includes a 'Discover' button, 'Connected State' and reports the number of discovered monitor (FSK) devices.  All Discoved 'monitor' devices are automatically linked to this new device as 'Connected devices'.  The README examples have also been updated to show how to link these to the board device using `via_device`.
* Home Assistant: Added 'Discover' button for the board device; this will perform a 10 second auto-scan and will update the device list via MQTT Discovery for all devices found.

### Fixed

* Added options to mqtt service for increased resilience of restarting the service upon failure [#62](https://github.com/Achronite/mqtt-energenie-ener314rt/issues/62)(@genestealer)
* Removed verbose logging of MQTT password [#66](https://github.com/Achronite/mqtt-energenie-ener314rt/issues/66)
* Only enable discovery if monitoring is also enabled [#82](https://github.com/Achronite/mqtt-energenie-ener314rt/issues/82)

### Changed

* **BREAKING DEPENDENCIES**: The version of mqtt.js has been updated, this newer version required node.js >= v18.2.0
* **BREAKING DEPENDENCIES**: The version of `energenie-ener314rt` needed is v0.7.x.  This uses a newer GPIO library that is compatible with the pi5. `gpiod` and `libgpiod` will need to be installed first
* Pretty printed all device JSON files
* MQTT Discovery: OEM Part Number and Device ID added to HA device model field (@genestealer)
* Submitting a cached command will now replace the existing cached command for the device
* Added data value to the reporting of `command` when it has been successfully cached (also only update MQTT when this is set to 0=None) [#69](https://github.com/Achronite/mqtt-energenie-ener314rt/issues/69),
    e.g. "Set Temperature" becomes "Set Temperature 18.5"
* Bumped MQTT to 5.3.5 [#62](https://github.com/Achronite/mqtt-energenie-ener314rt/issues/62)
* MQTT Discovery: In line with HA best practices, the primary entities for each of the FSK devices (see table below) have had their names updated to 'None'/null, e.g: sensor.motion_sensor.XXX.motion -> sensor.motion_sensor.XXX.  These entities are indicated by `"main":true` in the device json files.

|Device|Description|Primary Entity name set to null|
|---|---|---|
|MIHO004|MiHome Smart Monitor Plug (Pink)|real power|
|MIHO005|MiHome Smart Plug+ (Purple)|switch|
|MIHO006|MiHome House Monitor|apparent power|
|MIHO013|MiHome Radiator Valve|climate control|
|MIHO032|MiHome Motion sensor|motion|
|MIHO033|MiHome Open Sensor|contact|
|MIHO069|MiHome Heating Thermostat|climate control|
|MIHO089|MiHome Click|voltage|

See also: https://github.com/Achronite/energenie-ener314rt/releases/tag/v0.7.1  - Notably pi5 support and GPIO driver changes

<hr /><em>This discussion was created from the release <a href='https://github.com/Achronite/mqtt-energenie-ener314rt/releases/tag/v0.7.1'>v0.7.1</a>.</em>

**Comment by genestealer** [comment]
@Achronite that should be `sudo apt install gpiod libgpiod-dev` there is no libgpiod or am i missing something?
![image](https://github.com/Achronite/mqtt-energenie-ener314rt/assets/5207490/a40dd483-1847-42e4-8063-70ae69d041c8)

**Comment by Achronite** [comment]
No, I think you are right; I've got mixed up.  it is `libgpiod-dev` on raspbian.

**Comment by Achronite** [comment]
In the install instructions it already has it right, but i see it is confusing:
## Getting Started

1) Plug in your ENER314-RT-VER01 board from Energenie onto the 26 pin or 40 pin connector of your Raspberry Pi.

2) Install `gpiod` and `libgpiod` dependencies (as of v0.7.x) :
For example (Debian):
```
sudo apt-get install gpiod libgpiod-dev
